src/doc/playback_tests.cpp

```diff
@@ -500,6 +525,349 @@ TEST(Playback, OnePingPongInsideOther3)
   EXPECT_FALSE(play.isStopped());
 }
 
+// Special cases of OnePingPongInsideOther --------------------
+// When tagA->toFrame == tagB->toFrame == sprite->lastFrame
+// tagA->repeat == 2  and tagB->repeat == 2
+
+ TEST(Playback, OnePingPongInsideOther4)
+ {
+   //     A
+   // <------->
+   //       B
+   //     <--->
+   // 0 1 2 3 4
+
+   Tag* tagA = make_tag("A", 0, 4, AniDir::PING_PONG, 2);
+   Tag* tagB = make_tag("B", 2, 4, AniDir::PING_PONG, 2);
+   auto sprite = make_sprite(5, { tagA, tagB });
+
+   Playback play(sprite.get(), 0, Playback::Mode::PlayInLoop);
+   expect_frames(play, {0,1 , 2,3,4,3,2 , 1,0,
+                        0,1 , 2,3,4,3,2 , 1,0});
+   EXPECT_FALSE(play.isStopped());
+ }
+
+ TEST(Playback, OnePingPongInsideOther5)
+ {
+   //     A
+   // <------->
+   //       B
+   //     >---<
+   // 0 1 2 3 4
+   Tag* tagA = make_tag("A", 0, 4, AniDir::PING_PONG, 2);
+   Tag* tagB = make_tag("B", 2, 4, AniDir::PING_PONG_REVERSE, 2);
+   auto sprite = make_sprite(5, { tagA, tagB });
+
+   Playback play(sprite.get(), 0, Playback::Mode::PlayInLoop);
+   expect_frames(play, {0,1 , 4,3,2,3,4 , 1,0,
+                        0,1 , 4,3,2,3,4 , 1,0});
+   EXPECT_FALSE(play.isStopped());
+ }
+
+ TEST(Playback, OnePingPongInsideOther6)
+ {
+   //     A
+   // >-------<
+   //       B
+   //     <--->
+   // 0 1 2 3 4
+   Tag* tagA = make_tag("A", 0, 4, AniDir::PING_PONG_REVERSE, 2);
+   Tag* tagB = make_tag("B", 2, 4, AniDir::PING_PONG, 2);
+   auto sprite = make_sprite(5, { tagA, tagB });
+
+   Playback play(sprite.get(), 0, Playback::Mode::PlayInLoop);
+   expect_frames(play, {0,1 , 2,3,4,3,2 ,
+                        4,3,2,3,4 , 1,0,1 , 2,3,4,3,2 ,
+                        4,3,2,3,4 , 1,0,1 , 2,3,4,3,2});
+   EXPECT_FALSE(play.isStopped());
+ }
+
+TEST(Playback, OnePingPongInsideOther7)
+{
+  //     A
+  // >-------<
+  //       B
+  //     >---<
+  // 0 1 2 3 4
+  Tag* tagA = make_tag("A", 0, 4, AniDir::PING_PONG_REVERSE, 2);
+  Tag* tagB = make_tag("B", 2, 4, AniDir::PING_PONG_REVERSE, 2);
+  auto sprite = make_sprite(5, { tagA, tagB });
+
+  Playback play(sprite.get(), 0, Playback::Mode::PlayInLoop);
+  expect_frames(play, {0,1 , 4,3,2,3,4 ,
+                       2,3,4,3,2 , 1,0,1 , 4,3,2,3,4 ,
+                       2,3,4,3,2 , 1,0,1 , 4,3,2,3,4});
+  EXPECT_FALSE(play.isStopped());
+}
+
+// Special cases of OnePingPongInsideOther --------------------
+// When tagA->fromFrame == tagB->fromFrame == first frame
+// tagA->repeat == 2  and tagB->repeat == 2
+
+TEST(Playback, OnePingPongInsideOther8)
+{
+  //     A
+  // <------->
+  //   B
+  // <--->
+  // 0 1 2 3 4
+
+  Tag* tagA = make_tag("A", 0, 4, AniDir::PING_PONG, 2);
+  Tag* tagB = make_tag("B", 0, 2, AniDir::PING_PONG, 2);
+  auto sprite = make_sprite(5, { tagA, tagB });
+
+  Playback play(sprite.get(), 0, Playback::Mode::PlayInLoop);
+  expect_frames(play, {0,1,2,1,0 , 3,4,3 , 2,1,0,1,2 ,
+                       0,1,2,1,0 , 3,4,3 , 2,1,0,1,2 });
```

Comment:
I stop reviewing cases here, I think they are good so far. As we talked, we have two options: 1) just keep the most relevant cases (but don't create a lot of more cases, or at least if more cases are add some extra variations can be tested, like adding more repeats), or 2) group all these OnePingPongInsideOther in one test case with a `for()` to test more combinations (just if the expected sequence can be easily generated).